libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -87,6 +87,82 @@ proc init*(_: type[GossipSubParams]): GossipSubParams =
       maxNumElementsInNonPriorityQueue: DefaultMaxNumElementsInNonPriorityQueue
     )
 
+proc new*(
```

Comment:
the name `new` is for functions that introduce a `ref` in the return type - `GossipSubParams` - this should be called `init`: https://status-im.github.io/nim-style-guide/language.objconstr.html

Reply:
Done.